glass/fields.py

```diff
@@ -51,15 +51,25 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _R:
         return decorator
 
 
+def _is_inv_triangle_number(triangle_number: int) -> tuple[int, bool]:
```

Comment:
I'm not a big fan of this return signature. The signature minimises repetitions but I am open to any better implementations.

Reply:
Just have `_inv_triangle_number()` throw an exception if it's not -- both downstream functions can catch and rethrow it

Reply:
Thanks! I have updated the implementation.